src/fides/api/ctl/migrations/versions/48d9caacebd4_create_privacy_declarations_table.py

```diff
@@ -0,0 +1,166 @@
+            "updated_at",
+            sa.DateTime(timezone=True),
+            server_default=sa.text("now()"),
+            nullable=True,
+        ),
+        sa.Column("name", sa.String(), nullable=True),
+        sa.Column("data_categories", sa.ARRAY(sa.String()), nullable=True),
+        sa.Column("data_qualifier", sa.String(), nullable=True),
+        sa.Column("data_subjects", sa.ARRAY(sa.String()), nullable=True),
+        sa.Column("dataset_references", sa.ARRAY(sa.String()), nullable=True),
+        sa.Column("egress", sa.JSON(), nullable=True),
+        sa.Column("ingress", sa.JSON(), nullable=True),
+        sa.Column("system_id", sa.String(), nullable=False),
+        sa.Column("data_use", sa.String(), nullable=False),
+        sa.ForeignKeyConstraint(
+            ["data_use"],
+            ["ctl_data_uses.fides_key"],
+        ),
+        sa.ForeignKeyConstraint(
+            ["system_id"],
+            ["ctl_systems.fides_key"],
+            onupdate="CASCADE",
+            ondelete="CASCADE",
+        ),
+        sa.PrimaryKeyConstraint("id"),
+    )
+    op.create_index(
+        op.f("ix_privacydeclaration_data_use"),
+        "privacydeclaration",
+        ["data_use"],
+        unique=False,
+    )
+    op.create_index(
+        op.f("ix_privacydeclaration_id"), "privacydeclaration", ["id"], unique=False
+    )
+    op.create_index(
+        op.f("ix_privacydeclaration_system_id"),
+        "privacydeclaration",
+        ["system_id"],
+        unique=False,
+    )
+
+    bind = op.get_bind()
+    existing_declarations = bind.execute(
+        text("SELECT fides_key, privacy_declarations FROM ctl_systems;")
+    )
+    for row in existing_declarations:
+        system_id = row["fides_key"]
+        old_privacy_declarations = row["privacy_declarations"]
+        for privacy_declaration in old_privacy_declarations:
+            new_privacy_declaration_id: str = "pri_" + str(uuid.uuid4())
+            new_data = {
+                **privacy_declaration,
+                "system_id": system_id,
+                "id": new_privacy_declaration_id,
+            }
+
+            insert_privacy_declarations_query = text(
+                "INSERT INTO privacydeclaration (id, name, data_categories, data_qualifier, data_subjects, dataset_references, egress, ingress, system_id, data_use) "
+                "VALUES (:id, :name, :data_categories, :data_qualifier, :data_subjects, :dataset_references, :egress, :ingress, :system_id, :data_use)"
+            )
+
+            try:
+                bind.execute(
+                    insert_privacy_declarations_query,
+                    new_data,
+                )
+            except IntegrityError as exc:
+                raise Exception(
+                    f"Fides attempted to copy System.privacy_declarations into their own privacydeclaration rows but got error: {exc}. "
+                    f"Adjust the 'data_use' value to ponit to an existing data use to migrate successfully"
```

Comment:
ah yes this crossed my mind too! thanks for calling it out @pattisdr 

i definitely agree we should have better error handling in those cases. i think my only question is around whether we should scope it to this PR, or perhaps create a follow up ticket? i think the logic would need to be in the generic `crud` API or with an API to special-handle `DataUse` deletions, right? if that's the case, i'm tempted to split it out into a follow up - but if you're thinking it's really straightforward and could be tackled along with this, i'm happy to take a stab at it. let me know what you think @pattisdr!

Reply:
oh i guess this may relate to your [comment here](https://github.com/ethyca/fides/pull/3098#discussion_r1172837911) - if we cascade deletes from `DataUse`, this could get us what we want? although it would mean that a `DataUse` deletion would automatically delete privacy declarations that point to it - and i want to check with product (cc @mfbrown ) on whether that would be acceptable UX...

Reply:
Do you mean deleted from the taxonomy?

Reply:
@adamsachs re: cascade deletes I was talking about cascade update.  I meant with this unusual FK on data use, if the data use was updated, wouldn't we want it to update on the Privacy Declaration too?

Reply:
ok, so per conversation with @mfbrown, we can ticket out better handling of cascading taxonomy deletes as future work, but we'll keep behavior as it is currently in `main`, which is to _not_ prevent deletes of `DataUse`s that are associated with a `PrivacyDeclaration`. that means we'll need to break this FK link and just rely on a "soft" string match, as it stood before. i'll create the follow up ticket and make adjustments as needed.

Reply:
follow up issue here: https://github.com/ethyca/fides/issues/3119